main.ts

```diff
@@ -2,19 +2,17 @@ import { app, BrowserWindow, screen, Menu, nativeTheme, MenuItemConstructorOptio
 import * as path from 'path';
 import * as url from 'url';
 import { createMenuOptions } from './electron-utils/menu-bar';
+import { isDeveloperMode, isLinuxOs, isMacOs, isWindowsOs, appTitle } from './electron-utils/supporting-logic';
 import { getAccessToken } from './oauth';
 
 const Logger = require('electron-log');
 const ICON_PATH = path.join(__dirname, 'dist/favicon.512x512.png');
 
 let win: BrowserWindow = null;
-const args = process.argv.slice(1),
-  serve = args.some(val => val === '--serve');
+const isDevMode = isDeveloperMode();
 
-const isDevMode = !!serve;
-
-ipcMain.on('synchronous-message', (event, arg) => {
-  event.returnValue = process.platform === 'win32'
+ipcMain.on('synchronous-message', (event) => {
+  event.returnValue = isWindowsOs()
```

Comment:
Can we move this logic for finding the current directory into a separate function too?
So we'd end up with: `event.returnValue = getCurrentDirectory(isDevMode)`

Reply:
I've moved it such that now it is `event.returnValue = getCurrentDirectory(isWindowsOs, isDevMode)`.

Also, do you think we should move over the `windowOptions` chunk under `createWindow()` function to make it neater?

Reply:
Sure, we can move the window options into a separate module - but let's do this in a separate PR as it's not closely related to the changes here.

Reply:
Alright!